testdata/testoutput2

```diff
@@ -19612,6 +19612,693 @@ Failed: error -49 at offset 3 in replacement: unknown substring
+No match
+
+/[[A]||[B]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          cls:[B]
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    A
+ 0: A
+    B
+ 0: B
+\= Expect no match
+    C
+No match
+
+/[[^A][B]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[\x00-@B-\xff] (neg)
+          cls:[B]
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    B
+ 0: B
+    C
+ 0: C
+\= Expect no match
+    A
+No match
+
+/[^[A][B]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          cls:[B]
+          op: ||
+          op: ^
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    C
+ 0: C
+\= Expect no match
+    A
+No match
+    B
+No match
+
+/[^[A]&&[B]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          cls:[B]
+          op: &&
+          op: ^
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    A
+ 0: A
+    B
+ 0: B
+    C
+ 0: C
+
+/[A[]]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          cls:[\]]
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    A
+ 0: A
+    ]
+ 0: ]
+\= Expect no match
+    [
+No match
+
+/[A[^]]]/B,alt_extended_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          cls:[\x00-\^-\xff] (neg)
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    A
+ 0: A
+    [
+ 0: [
+    C
+ 0: C
+\= Expect no match
+    ]
+No match
+
+/[A[]]/B,alt_extended_class,allow_empty_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          cls:[]
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    A
+ 0: A
+\= Expect no match
+    ]
+No match
+    [
+No match
+
+/[A[^]]/B,alt_extended_class,allow_empty_class
+------------------------------------------------------------------
+        Bra
+        eclass[
+          cls:[A]
+          AllAny
+          op: ||
+        ]
+        Ket
+        End
+------------------------------------------------------------------
+    A
+ 0: A
+    C
+ 0: C
+    [
+ 0: [
+    ]
+ 0: ]
+
+/[A-C--B]/B,alt_extended_class
```

Comment:
Is there tests for `[A--B--C--D]` `[A--^B]` `[A---B]` `[A----B]` `[^--A]`.
Is there tests to reach the maximum depth?

Reply:
I have most of those tests.

The expression `[A--^B]` has a _literal_ ASCII `^` character, and so it's equivalent to simply `[A]`.

The expression `[A---B]` is a parse error. So are `[A----B]` `[^--A]`.

I have tests to reach maximum depth - a successful test at max depth, and a parse error test at max+1.